ReactWindows/ReactNative/Views/Text/ReactTextViewManager.cs

```diff
@@ -185,5 +187,203 @@ protected override RichTextBlock CreateViewInstance(ThemedReactContext reactCont
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactTextBlockSelectionEvent(
+                        view.GetTag(),
+                        selection));
+        }
+
+        /// <summary>
+        /// Returns a text representation of the selection in the RichTextBlock.
+        /// </summary>
+        /// <param name="view">The RichTextBlock.</param>
+        /// <returns>Text representation of the selection.</returns>
+        private string GetSelectedString(RichTextBlock view)
+        {
+            TextPointer selectionStart = view.SelectionStart;
+            TextPointer selectionEnd = view.SelectionEnd;
+
+            return GetStringByStartAndEndPointers(view, selectionStart, selectionEnd);
+        }
+
+        /// <summary>
+        /// Returns a text representation of the elements between <paramref name="start"/> and <paramref name="end"/> in the RichTextBlock.
+        /// </summary>
+        /// <param name="view">The RichTextBlock.</param>
+        /// <param name="start">Start TextPointer.</param>
+        /// <param name="end">End TextPointer.</param>
+        /// <returns></returns>
+        public static string GetStringByStartAndEndPointers(RichTextBlock view, TextPointer start, TextPointer end)
+        {
+            if (start == null || end == null || start == end)
+            {
+                return null;
+            }
+
+            // Parse the RichTextBlock contents.
+            StringBuilder selectedText = new StringBuilder();
+            Debug.Assert(view.Blocks.Count == 1, "RichTextBlock is expected to contain only one paragraph.");
+            Paragraph paragraph = view.Blocks.First() as Paragraph;
+            ProcessSelectedInlines(paragraph.Inlines, selectedText, false, start, end);
+
+            return selectedText.ToString();
+        }
+
+        /// <summary>
+        /// Recursively processes InlineCollection and builds text representation of the RichTextBlock selected content.
+        /// </summary>
+        /// <param name="inlines">InlineCollection to be processed.</param>
+        /// <param name="selectedText">Will contain the text representation upon procedure completion.</param>
+        /// <param name="accumulate">If 'true', includes inlines as part of the processed selection.</param>
+        /// <param name="selectionStart">TextPointer to selection start.</param>
+        /// <param name="selectionEnd">TextPointer to selection end.</param>
+        /// <returns>'true' if recursion terminal condition is met.</returns>
+        private static bool ProcessSelectedInlines(
+            InlineCollection inlines,
+            StringBuilder selectedText,
+            bool accumulate,
+            TextPointer selectionStart,
+            TextPointer selectionEnd)
+        {
+            foreach (Inline inline in inlines)
+            {
```

Comment:
Can inlines be null?

Reply:
Inline must be added to the collection as an instance, so it can't be null.

---
In reply to: [215225580](https://github.com/Microsoft/react-native-windows/pull/1951#discussion_r215225580) [](ancestors = 215225580)